src/server/game/AI/CreatureAI.cpp

```diff
@@ -134,7 +134,10 @@ void CreatureAI::MoveInLineOfSight(Unit* who)
         return;
 
     if (me->HasReactState(REACT_AGGRESSIVE) && me->CanStartAttack(who, false))
+    {
+        me->AddThreat(who, 0.0f); // ensure our initial target is the first thing added to threat list so we don't randomly switch off if DoZoneInCombat is called during the EnterCombat hook
```

Comment:
I think it should be added in Attack or in AttackStart (attack may return false)